src/codebuild-handler/index.ts

```diff
@@ -0,0 +1,86 @@
+import * as AWSLambda from 'aws-lambda';
+import * as AWS from 'aws-sdk';
+
+const cb = new AWS.CodeBuild();
+
+export async function onEventhandler(event: AWSLambda.CloudFormationCustomResourceEvent) {
+  switch (event.RequestType) {
+    case 'Create':
+    case 'Update':
+      return startBuild(event);
+    case 'Delete':
+      return onDelete();
+  }
+}
+
+export async function startBuild(event: AWSLambda.CloudFormationCustomResourceEvent) {
+  const requestType = event.RequestType;
+  const projectName = event.ResourceProperties.projectName;
+  if (requestType === 'Create' || requestType === 'Update') {
```

Comment:
💅 Afaict, `startBuild` is only called when `requestType` is one of these. This method can be simplified.